### ISSUE TITLE
Free Shipping Interface

### ISSUE THREAD

**Issue by marktnz1** [thirtybees/thirtybees#947]
Selecting Free Shipping under Shipping/Carriers/(Carrier)/"Shipping Location and costs" disables Ranges. This causes Free Shipping to be unusable in some cases and a minor UI inconvenience in other cases.

So when the user selects "Free Shipping"....

Allow selected Price Ranges to be >=0.00000 to infinity
and < to be 0.000000 to infinity

and 

Allow all or multiple Zones to be selected but disable the price and leave as 0.000000 (as its free shipping).

**Comment by Traumflug** [comment]
To be honest, I'm not happy with this _Free Shipping_ switch. As you kindly described, while it apparently works as designed, it adds more confusion than convenience.

Would removing this switch solve the problem for you? The ordering process could recognize free shipping by a shipping price of zero just as easily. Maybe it even does already, didn't check, yet.

**Comment by marktnz1** [comment]
Yes, I hear what you are saying and I mulled about that too. Back end it probably doesn't make any difference as its just adding a price of zero to the cost field (but that would have to be confirmed- hopefully the Free Shipping button doesnt do anything else than that). For user ease though, it would be nice to see a button that says Free Shipping and that just takes care of everything for them, instead of not being sure. I'd vote for the button staying and pushing it just adds zero. Then the only other thing the user has to do is select the zones and weight factors.

**Comment by Traumflug** [comment]
This "Free Shipping" feature is simply not designed to apply only to some zones or weight/price ranges. It's all-or-nothing.

Will change the topic of this issue to reflect outcome of the discussion. If you don't like it, please close the issue.

**Comment by marktnz1** [comment]
OK, sounds good, I'm happy with that

**Comment by marktnz1** [comment]
Note I currently get:

TECHNICAL ERROR:

Details:
Error thrown: [object Object]
Text status: parsererror

When I try to add the following:

This carrier is not free and the delivery announced is: Per arrangement.

The shipping cost is calculated according to the price and the tax rule No tax will be applied.

This carrier can deliver orders from 0.000000 $ to 1000 $. If the order is out of range, the behavior is to Apply the cost of the highest defined range.
This carrier will be proposed for those delivery zones

    Oceania

And it will be proposed for those client groups

    Visitor
    Guest
    Customer
